### PR TITLE
Fix for #20

### PR DESCRIPTION
**Pull request by phun-ky**
See #20

Basically, I edited `lib/dox-foundation.js` to pass one extra variable that defines current file we're rendering a template to, so we can figure out how deep the links should be.

``` js
  // How deep is your love?  
  if(file.currentFile.split('/').length > 1 ){
    targetLink = Array(file.currentFile.split('/').length).join('../') + file.targetFile;
  } else {
    targetLink = file.targetFile;
  }
```

### CODE REVIEW

**Comment by mattmcmanus** [comment]
I like where you are going with this. A couple things:
1. I brought in another pull request that fixed the windows support and for some reason made your PR conflict. Could you try updating?
2. I would leave the version number as it is. `npm version` will take care of incrementing when I publish it. [Docs](https://npmjs.org/doc/version.html)
3. Why did you add colors? I don't see it used anywhere
4. Please add yourself to the contributors in the package.json if you want
5. Finally, I added some comments to the code itself

File: lib/dox-foundation.js [hunk 1]
```diff
@@ -35,14 +36,22 @@ var templatePath = exports.templatePath = '../views/template.jade';
...
 
   file.dox.forEach(function(method){
     if (method.ctx) { names.push(method.ctx.name); }
   });
-  
-  return { source: file.sourceFile, target: file.targetFile, methods: names };
+
+  // How deep is your love?  
+  if(file.currentFile.split('/').length > 1 ){
+    targetLink = Array(file.currentFile.split('/').length).join('../') + file.targetFile;
```
Comment (mattmcmanus) [inline lib/dox-foundation.js]: This is how I was thinking of doing it. My only worry with this code is it's long term readability. It took me a minute to figure it out. Maybe just break it up?

```
var depth = file.currentFile.split('/').length,
      prefix = Array(depth).join('../');
targetLink =  prefix + file.targetFile;
```

**Comment by phun-ky** [comment]
1. Will try to update
2. Sorry, bad habit ;)
3. Again, a habit of mine when I debug node apps, I always add colors to add colors to output, it's brilliant when having 5x+ console.log statements. Will remove it ;)
4. Thanks man! Will do!
5. Will make it a bit more readable, I was just trying to make it tiny :)

**Comment by damienklinnert** [comment]
plz merge and update npm package

**Comment by phun-ky** [comment]
Finally. Added support for multiple directories aswell to this pull request.

I have not tested all scenarios, but I've tested with single directory and with two directories. Please test this PR and give me some feedback. 

You might want to refactor the code a bit.

**Comment by mattmcmanus** [comment]
Good golly. Thats a lot of work. Thanks so much @punk-ky. Since there is a lot here I want to make sure I give a good look over. Give me a couple days for that.

**Comment by phun-ky** [comment]
Take your time, I was in a bit of a rush to get this fixed, so couldn't test it properly. Let me know if there's something wrong.

**Comment by phun-ky** [comment]
Status on this? ;)

**Comment by mattmcmanus** [comment]
I'm really sorry Alex! If you can promise me that this code is good I'll just merge it in and deploy. It's been a busy couple weeks and I haven't had time to focus on this at all. I hate holding you up though. Once I get the final go ahead from you I'll send it out.

**Pull request merged**